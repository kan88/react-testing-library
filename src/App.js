import { Start } from './components/start/start.js';

function App() {
  return (
    <div className="App">
      <Start/>
    </div>
  );
}

export default App;
