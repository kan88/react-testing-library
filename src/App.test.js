import { fireEvent, render, screen } from '@testing-library/react';
// import userEvent from '@testing-library/user-event';
import App from './App';

describe('Testing react app', () => {
  test('Инпут, батон и хелло ворлд есть на странице', () => {
    render(<App />);
    const helloWorldElement = screen.getByText(/Hello world/i);
    const button = screen.getByRole('button')
    const input = screen.getByPlaceholderText(/input value/i)
    expect(helloWorldElement).toBeInTheDocument();
    expect(button).toBeInTheDocument();
    expect(input).toBeInTheDocument();
    expect(input).toMatchSnapshot();
  });
  
  test('отсутствие элемента с тесктом хелло', () => {
    render(<App />);
    const helloElement = screen.queryByText(/хелло/i);
    expect(helloElement).toBeNull();
  });

  test('асинхронный поиск через find', async () => {
    render(<App />);
    const objectElement = await screen.findByText(/Jest react testing/i);
    expect(objectElement).not.toBeNull();
    expect(objectElement).toBeInTheDocument();
    expect(objectElement).toHaveStyle({color: 'red'});
  });

  test('событие клика', async () => {
    render(<App />);
    const btn = screen.getByTestId('toggle__btn');
    expect(screen.queryByTestId('toggle__text')).toBeNull()
    fireEvent.click(btn)
    expect(screen.getByTestId('toggle__text')).toBeInTheDocument();
    fireEvent.click(btn)
    expect(screen.queryByTestId('toggle__text')).toBeNull();
  });

  test('событие инпута', async () => {
    render(<App />);
    const input = screen.getByPlaceholderText(/input value/i)
    expect(screen.queryByTestId('input__title')).toContainHTML('')
    //fireEvent искусственное событие userEvent более полная эмуляция
    fireEvent.input(input, {
      target: {value: '123'}
    })

    // userEvent.type(input, '123')
    expect(screen.queryByTestId('input__title')).toContainHTML('123')

  });

})

