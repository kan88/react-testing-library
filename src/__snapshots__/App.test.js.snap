// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing react app Инпут, батон и хелло ворлд есть на странице 1`] = `
<input
  placeholder="input value..."
  type="text"
  value=""
/>
`;
